src/features/DebugSession.ts

```diff
@@ -80,6 +80,12 @@ export class DebugSessionFeature implements IFeature {
                     }
                 }
             }
+            else if (config.script) {
+                // In this case, the user has explicitly defined a script path
+                // so make sure to set the cwd to that path if the cwd wasn't
+                // explicitly set
+                config.cwd = config.cwd || config.script;
```

Comment:
Weird, it was working fine for me.  But yeah, you're right...  In theory that would cause the workingDir to be set to the path of the file in the debug adapter.  Maybe we should take out this logic in one of the places?  Maybe make the debug adapter dumb and the extension code smart?